packages/lib/src/utils/useForm/useForm.ts

```diff
@@ -11,7 +11,7 @@ function useForm<FormSchema>(props: FormProps): Form<FormSchema> {
     /** Formats and validates a field */
     const processField = ({ key, value, mode }, fieldContext) => {
         // Find a formatting function either stored under 'key' or a level deeper under a 'formatter' property
-        const formatterFn = formatters?.[key]?.formatter ? formatters[key].formatter : formatters?.[key];
+        const formatterFn = (formatters as Formatter)?.[key]?.formatter ? (formatters as Formatter)[key].formatter : formatters?.[key];
```

Comment:
Is there any way to avoid the `as` here? This is error prone as we are telling Typescript which type that variable is. It should be the other way around.

I believe you could assert the type of this object using type-guard function